app/lib/actions/candlepin/owner/import.rb

```diff
@@ -1,18 +1,24 @@
 module Actions
   module Candlepin
     module Owner
-      class Import < Candlepin::Abstract
+      class Import < Candlepin::AbstractAsyncTask
         input_format do
           param :label
           param :path
           param :force
           param :upstream
         end
 
-        def run
+        def invoke_external_task
           options = input.slice(:force, :upstream)
-          output[:response] = ::Katello::Resources::Candlepin::Owner.import(input[:label], input[:path], options)
+          cp_response = JSON.parse(
+            ::Katello::Resources::Candlepin::Owner.import(input[:label], input[:path], options)
+          )
+          output[:task] = cp_response
+          output[:task_id] = cp_response['id']
+          poll_external_task # this return value sets external_task for future calls
```

Comment:
Sorry I'm jumping on late here. Just thinking back to when I was doing to katello agent stuff with polling which I later switched away from.

My method would have looked more like this:

```
       def invoke_external_task
          options = input.slice(:force, :upstream)
          ::Katello::Resources::Candlepin::Owner.import(input[:label], input[:path], options)
        end
```

If the other stuff is really needed then ignore this

Reply:
I think this was left over from when I was trying to figure out `Dynflow::Polling` and looking at https://github.com/Dynflow/dynflow/blob/e47118cd6bc6f972bec38844a6b06dd8b0739cca/lib/dynflow/action/polling.rb#L50

I said to myself, "Oh I better set `output[:task]`!"